frontend/libs/studio-components/src/hooks/useDebounce.ts

```diff
@@ -0,0 +1,25 @@
+import { useEffect, useRef, useCallback } from 'react';
+
+type UseDebounceOptions = {
+  debounceTimeInMs: number;
+};
+export const useDebounce = ({
+  debounceTimeInMs,
+}: UseDebounceOptions): { debounce: (callback: Function) => void } => {
+  const debounceRef = useRef<ReturnType<typeof setTimeout>>(undefined);
+  const debounce = useCallback(
+    (callback: Function): void => {
+      clearTimeout(debounceRef.current);
+      debounceRef.current = setTimeout(() => {
+        callback();
+      }, debounceTimeInMs);
+    },
+    [debounceTimeInMs],
+  );
```

Comment:
What about implementing support of local debounce time as well? 😊 It can simplify the code if we needs denounce two places within the same file with different timeouts.

```suggestion
  type DebounceOptions = {
      debounceTimeInMs: number;
  } 

  const debounce = useCallback(
    (callback: Function, { debounceTimeInMs: localDebounceTimeInMs}: DebounceOptions ): void => {
      clearTimeout(debounceRef.current);
      debounceRef.current = setTimeout(() => {
        callback();
      }, localDebounceTimeInMs || debounceTimeInMs);
    },
    [debounceTimeInMs],
  );
```

Reply:
Good suggestion! Committed but changed debounceOptions to be optional 👍